crates/net/dns/src/config.rs

```diff
@@ -0,0 +1,23 @@
+use std::time::Duration;
+
+/// Settings for the [DnsDiscoveryClient](crate::DnsDiscoveryClient).
+#[derive(Debug, Clone)]
+pub struct DnsDiscoveryConfig {
+    /// Timeout for DNS lookups.
+    ///
+    /// Default: 5s
+    pub lookup_timeout: Duration,
+    /// The rate at which lookups should be re-triggered.
+    ///
+    /// Default: 30min
+    pub lookup_interval: Duration,
```

Comment:
Is this the lookup on the root node or to walk the entries in the tree?

Reply:
this will be the interval to revisit trees